src/main/kotlin/io/github/inductiveautomation/kindling/log/LoggerNamePanel.kt

```diff
@@ -0,0 +1,285 @@
+import io.github.inductiveautomation.kindling.utils.collapseAll
+import io.github.inductiveautomation.kindling.utils.expandAll
+import io.github.inductiveautomation.kindling.utils.getAll
+import io.github.inductiveautomation.kindling.utils.selectAll
+import net.miginfocom.swing.MigLayout
+import javax.swing.JButton
+import javax.swing.JComponent
+import javax.swing.JMenuItem
+import javax.swing.JPanel
+import javax.swing.JPopupMenu
+import javax.swing.JScrollPane
+import javax.swing.tree.TreePath
+
+class LoggerNamePanel(private val rawData: List<LogEvent>) : FilterPanel<LogEvent>(), PopupMenuCustomizer {
+    private val isTreeAvailable = rawData.first() is SystemLogEvent
+
+    private var isTreeMode: Boolean = ShowLogTree.currentValue && isTreeAvailable
+        set(value) {
+            isContextChanging = true
+
+            maintainSelection(value)
+            field = value
+
+            mainTreeComponent.isVisible = value
+            mainListComponent.isVisible = !value
+
+            isContextChanging = false
+        }
+
+    private var isContextChanging: Boolean = false
+
+    private val filterList = FilterList(toStringFn = ::getSortKey).apply {
+        setModel(
+            FilterModel(
+                rawData.groupingBy(LogEvent::logger).eachCount(),
+                ::getSortKey,
+            ),
+        )
+        selectAll()
+
+        // Right clicking will check and uncheck an item. We don't want that for this list.
+        isClickInCheckBoxOnly = true
+
+        ShowFullLoggerNames.addChangeListener {
+            model = model.copy(comparator)
+        }
+
+        checkBoxListSelectionModel.addListSelectionListener { e ->
+            if (!e.valueIsAdjusting && !isContextChanging) {
+                listeners.getAll<FilterChangeListener>().forEach(FilterChangeListener::filterChanged)
+            }
+        }
+
+        attachPopupMenu {
+            JPopupMenu().apply {
+                val prompt = if (ShowFullLoggerNames.currentValue) "Show Compact Names" else "Show Full Names"
+                add(
+                    JMenuItem(
+                        Action(prompt) {
+                            ShowFullLoggerNames.currentValue = !ShowFullLoggerNames.currentValue
+                        },
+                    ),
+                )
+            }
+        }
+    }
+
+    private val sortButtons = filterList.createSortButtons()
+
+    @Suppress("UNCHECKED_CAST")
+    private val filterTree: LogTree = LogTree(
+        if (isTreeAvailable) {
+            rawData as List<SystemLogEvent>
+        } else {
+            emptyList()
+        },
+    ).apply {
+        selectionModel = NoSelectionModel()
+        checkBoxTreeSelectionModel.addTreeSelectionListener {
+            if (!isContextChanging) {
+                listeners.getAll<FilterChangeListener>().forEach(FilterChangeListener::filterChanged)
+            }
+        }
+    }
+
+    private val currentSelectedLeafNodes: List<String>
```

Comment:
`get()` is evaluated for each `LogEvent` on filter change. Can be optimized to run once on logger change and then re-used for all `LogEvents`.

Reply:
Ah, good catch. I'll definitely fix this before merge.